### PR TITLE
Adds redirection for metadata summary page & page titles

### PR DESCRIPTION
**Pull request by antony-britto**
1) Adds redirection between Ng & react pages of metadata summary.
2) Adds page titles for metadata pages.

### CODE REVIEW

File: app/cdap/components/Metadata/SearchSummary/EntityTopBar/index.tsx [hunk 1]
```diff
@@ -55,16 +57,35 @@ const EntityTab = styled(Tab)`
...
+
+  if (redirectUrl) {
+    return <Redirect to={redirectUrl} />;
+  }
 
-  function handleTabChange(event, newValue: string) {
-    setEntityTypeTab(Number(newValue));
+  function handleTabChange(event, newValue: number) {
+    if (defaultTab !== newValue) {
+      const baseUrl = `/ns/${getCurrentNamespace()}/metadata/${entityType}/${entityId}/`;
```
Comment (njbriggs) [inline app/cdap/components/Metadata/SearchSummary/EntityTopBar/index.tsx:82]: Is there any way to leverage the router so we don't need to put URLs in the component?

File: app/tracker/routes.js [hunk 2]
```diff
@@ -145,7 +147,15 @@ angular.module(PKG.name + '.feature.tracker')
             controller: 'TrackerMetadataController',
             controllerAs: 'MetadataController',
             onEnter: function($stateParams) {
-              document.title = `${productName} | Search | ${$stateParams.entityId} | Summary`;
+              // Redirect to react page when the feature is turned on
+              if (window.CaskCommon.ThemeHelper.Theme.isMetadataInReact) {
+                let searchUrl = reactAppUrl.summary.replace('<namespace>', $stateParams.namespace);
```
Comment (njbriggs) [inline app/tracker/routes.js:152]: Is there any way to use router to avoid manual URL manipulation here?

Comment (antony-britto) [reply app/tracker/routes.js:152]: This is angular app's router. We need to construct the react url and redirect. Couldn't find other way.

Comment (njbriggs) [reply app/tracker/routes.js:152]: Arrrggghhh good point.

**Review by njbriggs — APPROVED** [review]

**Pull request merged**